### DISCUSSION TITLE
Do sourcemaps generated by ts-node always have an empty names array?

### DISCUSSION THREAD

**Discussion by moltenice** [TypeStrong/ts-node#1710] [Q&A]
`tsc` currently has an issue with sourcemaps having an empty names array: https://github.com/microsoft/TypeScript/issues/9627

This causes issues when debugging because Typescript variables sometimes get renamed when compiling to Javascript and so referencing them when debugging will fail unless you reference them with their compiled name.

My understanding is that `ts-node` uses `tsc` under the hood so would suffer from the same issue.

Is that right? Do `ts-node` sourcemaps always have an empty names array?

**Comment by cspotcode** [comment]
When using our SWC mode, we use SWC under the hood.  So that might be different.

Check out our website, we have a page dedicated to "SWC" which explains how to enable it.

**Reply by moltenice** [reply]
Same issue with "SWC" unfortunately :( checked with Webstorm
